### ISSUE TITLE
SD mount and read fail on 2.0.3 idf 4.4.1

### ISSUE THREAD

**Issue by tsctrl** [espressif/arduino-esp32#6743]
### Board

ESP32 Dev Module

### Device Description

devkit

### Hardware Configuration

// GPIO15 SCK
// GPIO19 MISO
// GPIO23 MOSI
// SD mount pin=22

### Version

v2.0.3

### IDE Name

IDF Component, Arduino IDE

### Operating System

Windows 10

### Flash frequency

40MHz

### PSRAM enabled

no

### Upload speed

115200

### Description

on old library. i have to take chance and if fortunate the sd will be detected and load files. If not i have to pull out and in the SD during the startup. There is no pull up, the sd are connected directly to the SPI and cs pin. No other peripheral are connected.

same code on my end, I am getting error as below on my sd for 2.0.3:
```
Debug: SD mount pin: 22
[ 20535][E][sd_diskio.cpp:126] sdSelectCard(): Select Failed
[ 20535][E][sd_diskio.cpp:789] sdcard_mount(): f_mount failed: (3) The physical drive cannot work
[ 20837][E][sd_diskio.cpp:126] sdSelectCard(): Select Failed
Debug: SD Card Mount Failed!
```

basically no chances for the sd card to mount at all.
anyone can confirm sd are working on your end? or anyone have similar issue?

### Sketch

```cpp
-
```

### Debug Message

```plain
as above
```

### Other Steps to Reproduce

-

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

**Comment by tsctrl** [comment]
ground pin in module causing issue. changing to another gnd fix this and sd are loaded successfullly
